src/Sentry/Platforms/iOS/Extensions/CocoaExtensions.cs

```diff
@@ -1,8 +1,110 @@
+using Sentry.Extensibility;
+using SentryCocoa;
+
 namespace Sentry.iOS.Extensions;
 
 internal static class CocoaExtensions
 {
     public static DateTimeOffset ToDateTimeOffset(this NSDate timestamp) => new((DateTime)timestamp);
 
     public static NSDate ToNSDate(this DateTimeOffset timestamp) => (NSDate)timestamp.UtcDateTime;
+
+    public static string? ToJsonString(this NSObject? obj, IDiagnosticLogger? logger = null)
+    {
+        if (obj == null)
+        {
+            return null;
+        }
+
+        if (obj is ISentrySerializable serializable)
+        {
+            // For types that implement Sentry Cocoa's SentrySerializable protocol (interface),
+            // We should call that first, and then serialize the result to JSON later.
+            obj = serializable.Serialize();
+        }
+
+        // Now we will use Apple's JSON Serialization functions.
+        // See https://developer.apple.com/documentation/foundation/nsjsonserialization
+
+        if (!NSJsonSerialization.IsValidJSONObject(obj))
+        {
+            logger?.LogWarning("Cannot serialize a {0} directly to JSON", obj.GetType().Name);
+            return null;
+        }
+
+        try
+        {
+            using var data = NSJsonSerialization.Serialize(obj, 0, out _);
+            return data.ToString();
+        }
+        catch (Exception ex)
+        {
+            logger?.LogError("Error serializing {0} to JSON", ex, obj.GetType().Name);
+            return null;
+        }
+    }
+
+    public static Dictionary<string, string> ToStringDictionary<TValue>(
+        this NSDictionary<NSString, TValue>? dict,
+        IDiagnosticLogger? logger = null)
+        where TValue : NSObject
+    {
+        if (dict == null)
+        {
+            return new Dictionary<string, string>();
+        }
+
+        var result = new Dictionary<string, string>(capacity: (int)dict.Count);
+        foreach (var key in dict.Keys)
+        {
+            var value = dict[key];
```

Comment:
```suggestion
        foreach (var keyValue in dict)
        {
            var value = keyValue.Value;
            var key = keyValue.Key;
```

no need to do a dictionary lookup